### ISSUE TITLE
Troubleshooting Assistance: Failed to find a supported JDK

### ISSUE THREAD

**Issue by chenry22** [NationalSecurityAgency/ghidra#5763]
I know there are other closed threads on this same problem, but none of the solutions provided there have helped me.I've been trying to open Ghidra after downloading version 10.3.3 and I keep getting met with the "Failed to find a supported JDK" error message. I've tried to fix it by adding to my PATH and JAVA_HOME environment variables, but nothing has seemed to work so far...

I'm using a MacBook Air w/ an Apple M1 chip and trying to run Ghidra 10.3.3 with Java 17.0.8. Also I'm using a zsh shell instead of bash...

Currently these are the outputs I'm getting:

echo $PATH ->
/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin: ... 
[this is the first thing included in the PATH variable]

echo $JAVA_HOME ->
/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home

java -version ->
openjdk version "17.0.8.1" 2023-08-24
OpenJDK Runtime Environment Temurin-17.0.8.1+1 (build 17.0.8.1+1)
OpenJDK 64-Bit Server VM Temurin-17.0.8.1+1 (build 17.0.8.1+1, mixed mode)

Any advice on fixing this is much appreciated...

**Comment by dev747368** [comment]
Can you tried with a bash shell?

**Comment by chenry22** [comment]
Same issue, even with environment variables set the same way

**Comment by ryanmkurtz** [comment]
* Are you using an unaltered download of 10.3.3 from the Releases section of our Github page?
* When you launched Ghidra, did it ever prompt you to select the location of your JDK?
* Did you download the x64 or aarch64 version of the Temurin JDK?
* Any chance you installed the JRE instead of the JDK?

**Comment by chenry22** [comment]
Yes, I am using the download directly from the GitHub releases page. I've tried with other release versions too (specifically ones with a lower/different Java version requirement) but ran into the same issues.

Ghidra has never prompted me for the location of the JDK whenever I've run the executable ghidraRun.

I downloaded the aarch64 version of Temurin JDK since I'm working with an Apple M1 chip and I definitely installed the JDK.

To answer your second question more specifically, this is the output when I run the executable in my terminal:

**./ghidraRun 
Error: Could not find or load main class LaunchSupport
Caused by: java.lang.ClassNotFoundException: LaunchSupport
Error: Could not find or load main class LaunchSupport
Caused by: java.lang.ClassNotFoundException: LaunchSupport
Error: Could not find or load main class LaunchSupport
Caused by: java.lang.ClassNotFoundException: LaunchSupport

Failed to find a supported JDK.  Please refer to the Ghidra Installation Guide's Troubleshooting section.**

**Comment by ryanmkurtz** [comment]
I think you may have downloaded just the source repository, and not the built release (which is under Assets).

**Comment by chenry22** [comment]
Ok I think I am a little confused then, what do I need to download?

**Comment by chenry22** [comment]
So sorry for all the hassle, I ended up fixing it on my own. 

All I had to do was read further down in the Installation Guide to realize I needed to do a couple extra steps to build the required native binaries on an Apple Silicon processor. After doing this everything seems to be working fine. Again, really sorry for making a whole post and everything for something I could've solved with just the materials provided by you guys, but thank you for trying to help!
